src/renderer.cc

```diff
@@ -148,17 +277,13 @@ static void draw_triangle(const GLView::shaderinfo_t *shaderinfo, const Vector3d
 			glVertex3f(p2[0], p2[1], p2[2] + z);
 		}
 		break;
-	case GLView::shaderinfo_t::NONE:
-	case GLView::shaderinfo_t::SELECT_RENDERING:
-		if (mirror) {
-			glVertex3f(p0[0], p0[1], p0[2] + z);
-			glVertex3f(p2[0], p2[1], p2[2] + z);
-			glVertex3f(p1[0], p1[1], p1[2] + z);
-		} else {
-			glVertex3f(p0[0], p0[1], p0[2] + z);
-			glVertex3f(p1[0], p1[1], p1[2] + z);
-			glVertex3f(p2[0], p2[1], p2[2] + z);
+	default:
+	case Renderer::SELECT_RENDERING:
+		glVertex3d(p0[0], p0[1], p0[2] + z);
+		if (!mirror) {
+			glVertex3d(p1[0], p1[1], p1[2] + z);
```

Comment:
I cannot produce a bug with it, but it sounds to me like only 2 vertices in case of `mirror=true` should be a bug

Reply:
I believe this was an error introduced during merge. The `if(mirror)` case needs to be added back in. Most likely, right click mouse select is not working correctly in immediate rendering mode because of this.